agent/config/types.go

```diff
@@ -294,4 +294,12 @@ type Config struct {
 
 	// CgroupCPUPeriod is config option to set different CFS quota and period values in microsecond, defaults to 100 ms
 	CgroupCPUPeriod time.Duration
+
+	// SpotInstanceDrainingEnabled, if true, agent will poll the container instance's metadata endpoint for an ec2 spot
+	//   instance termination notice. If EC2 sends a spot termination notice, then agent will set the instance's state
```

Comment:
nit: agent is not actually the one setting the instance state as DRAINING. Should we mention backend side is the one transiting the state to DRAINING here?

Reply:
Agent calls the API that sets the state to DRAINING though